### ISSUE TITLE
Make `supported_by_runtime_system()` constexpr

### ISSUE THREAD

**Issue by anonrig** [simdutf/simdutf#569]
Starting from C++20, we can make `supported_by_runtime_system()` function `constexpr`. The limitations are:

```
/home/yagiz/coding/simdutf/include/simdutf/internal/isadetection.h:195:3: warning: ‘asm’ in ‘constexpr’ function only available with ‘-std=c++20’ or ‘-std=gnu++20’ [-Wc++20-extensions]
  195 |   asm volatile("cpuid\n\t" : "+a"(a), "=b"(b), "+c"(c), "=d"(d));
      |   ^~~
/home/yagiz/coding/simdutf/include/simdutf/internal/isadetection.h:181:23: warning: invalid return type ‘void’ of ‘constexpr’ function ‘constexpr void simdutf::internal::cpuid(uint32_t*, uint32_t*, uint32_t*, uint32_t*)’ [-Winvalid-constexpr]
  181 | static constexpr void cpuid(uint32_t *eax, uint32_t *ebx, uint32_t *ecx,
```

**Comment by lemire** [comment]
I don't think it can be a compile-time constant because it is literally evaluated at runtime. :-/ It says so in the name. :-)

**Comment by lemire** [comment]
It could be conditionally compile-time... that is, on some systems.

Or we could have compile-time versions of our functions.

I am not closing this because the issue raised by @anonrig is valid. It would be cool that everything was compile-time evaluated when possible and it is 100% possible to do so.

**Comment by lemire** [comment]
I am inviting contributions. This is not impossibly hard.

**Comment by WojciechMula** [comment]
IMHO, the only way to make that method constexpr would be compile the library with some options, like SIMDUTF_FORCE_X86_ICELAKE and then discard any runtime checks. Assuming this kind of options is provided by an external build system/a person who knows what they're doing.

**Comment by lemire** [comment]
@WojciechMula In general, you are correct, but in specific cases it is doable. So if you are a 64-bit ARM system, you can make it `constexpr`.

**Comment by lemire** [comment]
`supported_by_runtime_system()` is a runtime query, so in general it can't be constexpr. What this issue was really after (evaluating simdutf at compile time) is tracked in #865, where almost everything is now constexpr. #891 and PR #920 cover the rest. Closing as a duplicate of #865.
